ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/PostCommitValidation.scala

```diff
@@ -122,29 +122,23 @@ private[appendonlydao] object PostCommitValidation {
         transaction: CommittedTransaction,
         divulged: Set[ContractId],
     ): Set[ContractId] = {
-      val (createdInTransaction, referred) =
-        transaction.fold((Set.empty[ContractId], Set.empty[ContractId])) {
-          case ((created, ids), (_, c: Create)) =>
-            (created + c.coid, ids)
-          case ((created, ids), (_, e: Exercise)) if !divulged(e.targetCoid) =>
-            (created, ids + e.targetCoid)
-          case ((created, ids), (_, f: Fetch)) if !divulged(f.coid) =>
-            (created, ids + f.coid)
-          case ((created, ids), (_, l: LookupByKey)) =>
-            (created, l.result.filterNot(divulged).fold(ids)(ids + _))
-          case ((created, ids), _) => (created, ids)
-        }
-      referred.diff(createdInTransaction)
+      transaction.inputContracts.diff(divulged)
```

Comment:
If this is a one-liner used only in one place I'd suggest to just inline it.

Reply:
I considered it but it seemed easier for review at least to only replace the implementation but leave it in a separate definition. Don’t feel strongly though so happy to inline it.

Reply:
Good call as a first step. 🙇🏻